pkg/controller/filter.go

```diff
@@ -35,6 +38,7 @@ type Filter interface {
 type MastershipFilter struct {
 	Store    mastershipstore.Store
 	Resolver DeviceResolver
+	NodeID   cluster.NodeID
```

Comment:
We should probably make this private and add a `New` function to force it to be set correctly. Feels wrong to force other packages to set the node ID here.

Reply:
@kuujo 
Agreed. Let me fix it.

Reply:
I added a NewMastershipFilter function which accepts three arguments. The last argument is nodeID which can be empty and when it is empty it will be set to cluster NodeID otherwise it will be set to whatever it is passed to it (like what we have in filter_test.go). I also added a GetNodeID to filter that I can retrieve the private field in Accept function for checking local cluster node.